app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -381,13 +381,7 @@ def get_content_view_environments(label = nil, organization = nil)
           where("#{Organization.table_name}.id = ?", organization.id)
       environments = environments.where("#{Katello::ContentViewEnvironment.table_name}.label = ?", label) if label
 
-      environments.delete_if do |env|
-        if env.content_view.default
-          !env.environment.readable?
-        else
-          !env.content_view.readable?
-        end
-      end
+      environments = environments.to_a.delete_if { |env| (env.content_view.default && !env.environment.readable?) || !env.content_view.readable? }
```

Comment:
You probably need to cast environments to an array before you manipulate it with array methods:
```ruby
> users = User.all
#<ActiveRecord::Relation [#<User id: 1...
> users.count
10
> users.to_a.delete_if { |user| user.id.even? }
#<ActiveRecord::Relation [#<User id: 1...
> users.to_a.count
5
> users.count
10
```